### PR TITLE
Shutdown gracefully via signal interrupt

### PR DESCRIPTION
**Pull request by hasyimibhar**
Issue: https://github.com/omshub/core-api/issues/6

Also removed hardcoded timeout for shutdown (5 seconds), and added `DefaultConfig` for sensible defaults. In the future we should use struct tags for defaults (should be covered by https://github.com/omshub/core-api/issues/11).

### CODE REVIEW

File: internal/api/config.go [hunk 1]
```diff
@@ -1,5 +1,16 @@
 package api
 
+import "time"
+
 type Config struct {
-	Port string
+	Port             string
+	ShutdownDuration time.Duration
```
Comment (disposedtrolley) [inline internal/api/config.go]:
```suggestion
	ShutdownTimeout time.Duration
```

WDYT?

**Review by disposedtrolley — DISMISSED** [review]
LGTM! A small comment on config naming, but happy either way.

File: internal/api/config.go (hunk 1 above)
Comment (hasyimibhar) [reply internal/api/config.go]: Timeout sounds better. Let me change it.

**Review by disposedtrolley — DISMISSED** [review]

**Review by awpala — DISMISSED** [review]

**Comment by disposedtrolley** [comment]
This is a little annoying (but sensible), but GH restricts use of repository secrets in workflows started in a fork. This is why the Terraform pipeline is failing.

I think you're in the OMShub org now @hasyimibhar. Can I trouble you to open a PR from a branch within the org's repo?

**Comment by hasyimibhar** [comment]
Recreated here: https://github.com/omshub/core-api/pull/25

**Pull request closed without merging**